File: src/main/java/com/itsci/mju/maebanjumpen/controller/PartyRoleController.java
// File: src/main/java/com/itsci/mju/maebanjumpen/controller/PartyRoleController.java (No Change)
package com.itsci.mju.maebanjumpen.controller;

import com.itsci.mju.maebanjumpen.dto.PartyRoleDTO;
import com.itsci.mju.maebanjumpen.service.PartyRoleService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/maeban/party-roles")
@RequiredArgsConstructor
public class PartyRoleController {

    private final PartyRoleService partyRoleService;

    @GetMapping
    public ResponseEntity<List<PartyRoleDTO>> getAllPartyRole() {
        List<PartyRoleDTO> partyRoles = partyRoleService.getAllPartyRoles();
        return ResponseEntity.ok(partyRoles);
    }

    @GetMapping("/{id}")
    public ResponseEntity<PartyRoleDTO> getPartyRole(@PathVariable int id) {
        PartyRoleDTO partyRole = partyRoleService.getPartyRoleById(id);
        if (partyRole != null) {
            return ResponseEntity.ok(partyRole);
        }
        return ResponseEntity.notFound().build();
    }

    @PostMapping
    public ResponseEntity<PartyRoleDTO> createPartyRole(@RequestBody PartyRoleDTO partyRoleDto) {
        try {
            PartyRoleDTO savedPartyRole = partyRoleService.savePartyRole(partyRoleDto);
            return ResponseEntity.status(HttpStatus.CREATED).body(savedPartyRole);
        } catch (IllegalArgumentException e) {
            return ResponseEntity.badRequest().body(null);
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
        }
    }

    @PutMapping("/{id}")
    public ResponseEntity<PartyRoleDTO> updatePartyRole(@PathVariable int id, @RequestBody PartyRoleDTO partyRoleDto) {
        try {
            PartyRoleDTO updated = partyRoleService.updatePartyRole(id, partyRoleDto);
            if (updated != null) {
                return ResponseEntity.ok(updated);
            }
            return ResponseEntity.notFound().build();
        } catch (IllegalArgumentException e) {
            return ResponseEntity.badRequest().body(null);
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
        }
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deletePartyRole(@PathVariable int id) {
        partyRoleService.deletePartyRole(id);
        return ResponseEntity.noContent().build();
    }
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/partyrole/controller/PartyRoleController.kt
package com.itsci.mju.maebanjumpen.partyrole.controller

import com.itsci.mju.maebanjumpen.partyrole.dto.PartyRoleDTO
import com.itsci.mju.maebanjumpen.partyrole.service.PartyRoleService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/maeban/party-roles")
class PartyRoleController(private val partyRoleService: PartyRoleService) {

    @GetMapping
    fun getAllPartyRole(): ResponseEntity<List<PartyRoleDTO>> {
        val partyRoles = partyRoleService.getAllPartyRoles()
        return ResponseEntity.ok(partyRoles)
    }

    @GetMapping("/{id}")
    fun getPartyRole(@PathVariable id: Int): ResponseEntity<PartyRoleDTO> {
        val partyRole = partyRoleService.getPartyRoleById(id)
        return if (partyRole != null) {
            ResponseEntity.ok(partyRole)
        } else {
            ResponseEntity.notFound().build()
        }
    }

    @PostMapping
    fun createPartyRole(@RequestBody partyRoleDto: PartyRoleDTO): ResponseEntity<PartyRoleDTO?> {
        return try {
            val savedPartyRole = partyRoleService.savePartyRole(partyRoleDto)
            ResponseEntity.status(HttpStatus.CREATED).body(savedPartyRole)
        } catch (e: IllegalArgumentException) {
            ResponseEntity.badRequest().body(null)
        } catch (e: Exception) {
            ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null)
        }
    }

    @PutMapping("/{id}")
    fun updatePartyRole(@PathVariable id: Int, @RequestBody partyRoleDto: PartyRoleDTO): ResponseEntity<PartyRoleDTO?> {
        return try {
            val updated = partyRoleService.updatePartyRole(id, partyRoleDto)
            if (updated != null) {
                ResponseEntity.ok(updated)
            } else {
                ResponseEntity.notFound().build()
            }
        } catch (e: IllegalArgumentException) {
            ResponseEntity.badRequest().body(null)
        } catch (e: Exception) {
            ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null)
        }
    }

    @DeleteMapping("/{id}")
    fun deletePartyRole(@PathVariable id: Int): ResponseEntity<Void> {
        partyRoleService.deletePartyRole(id)
        return ResponseEntity.noContent().build()
    }
}

